src/transforms/aggregate.js

```diff
@@ -215,20 +215,34 @@ exports.calcTransform = function(gd, trace, opts) {
     var groupArray = Lib.getTargetArray(trace, {target: groups});
     if(!groupArray) return;
 
-    var i, vi, groupIndex;
+    var i, vi, groupIndex, newGrouping;
 
     var groupIndices = {};
+    var indexToPoints = {};
     var groupings = [];
+
+    var prevTransforms = trace.transforms.filter(function(tr) {return tr.indexToPoints;});
+    var originalPointsAccessor = prevTransforms.length ?
+        function(i) {return prevTransforms[prevTransforms.length - 1].indexToPoints[i];} :
+        function(i) {return [i];};
+
     for(i = 0; i < groupArray.length; i++) {
         vi = groupArray[i];
         groupIndex = groupIndices[vi];
         if(groupIndex === undefined) {
             groupIndices[vi] = groupings.length;
-            groupings.push([i]);
+            newGrouping = [i];
+            groupings.push(newGrouping);
+            indexToPoints[groupIndices[vi]] = originalPointsAccessor(i);
+        }
+        else {
+            groupings[groupIndex].push(i);
+            indexToPoints[groupIndices[vi]] = indexToPoints[groupIndices[vi]].concat(originalPointsAccessor(i));
         }
-        else groupings[groupIndex].push(i);
     }
 
+    opts.indexToPoints = indexToPoints;
```

Comment:
Since this isn't a real attribute but we're putting it back into `fullData` lets give it an `_` - ie `opts._indexToPoints`.